src/figuresClasses.ts

```diff
@@ -1,19 +1,80 @@
-export interface Figure {
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  public shape: Shape = 'triangle';
 
-}
+  constructor(
+    public color: Color,
+    public sideA: number,
+    public sideB: number,
+    public sideC: number,
+  ) {
+    if (sideA <= 0 || sideB <= 0 || sideC <= 0) {
+      throw new Error('No one side cannot be less than or equal to 0');
+    }
+
+    const sides: number[] = [sideA, sideB, sideC].sort(
+      (current, next) => next - current,
+    );
+
```

Comment:
would be nice to move this logic to a separate method in the constructor

Reply:
packed up everything into validation function (method), please, check if it is ok